icu4c/source/common/putil.cpp

```diff
@@ -1361,7 +1361,7 @@ uprv_pathIsAbsolute(const char *path)
 
 /* Backup setting of ICU_DATA_DIR_PREFIX_ENV_VAR
    (needed for some Darwin ICU build environments) */
-#if U_PLATFORM_IS_DARWIN_BASED && TARGET_OS_SIMULATOR
+#if U_PLATFORM_IS_DARWIN_BASED && defined(TARGET_OS_SIMULATOR) && TARGET_OS_SIMULATOR
```

Comment:
In theory this could have been
```
#if U_PLATFORM_IS_DARWIN_BASED && (defined(TARGET_OS_SIMULATOR) && TARGET_OS_SIMULATOR) || (defined(TARGET_IPHONE_SIMULATOR) && TARGET_IPHONE_SIMULATOR)
```
but I doubt that it makes much sense to compile the latest ICU for iPhone on those ancient OSes.

(Not that I know what this code is good for or that I bothered checking when it became deprecated; the above comment is pure speculation.)